### DISCUSSION TITLE
Allow floor to be built under furniture/other stuff

### DISCUSSION THREAD

**Discussion by adiamaq** [DFHack/dfhack#3053] [Ideas]
I'm facing the situation where I want to renovate a floor on a room which has already some stuff like beds or chairs
![image](https://user-images.githubusercontent.com/31577413/226283365-00c170a0-b2e6-4876-ba2c-5c0fde3a1ce8.png)
Is there any way for DFHack to force the update of flooring tiles under these?

**Comment by tily-lass** [comment]
oh, this would be so amazing

**Comment by JuanCobas** [comment]
THIS needs to be done, It's sooo annoying have to remove furniture to put a floor.

**Comment by adiamaq** [comment]
@myk002 Do you think this will be possible to do at some point?
Basically, ignore if there's an object type furniture when you're placing floors. I know it's not realistic but would be such a QoL not having to dismantle a whole room =)

**Comment by myk002** [comment]
It might be possible to "magic" a floor into place or magically change the material of an existing floor, but there is no way to *designate* that you want a floor installed such that DF will actually assign a dwarf to do it. One reason for this (there may be more) is that there can only be one building per tile, and a planned floor is a building (as far as DF is concerned) until it becomes a built construction. The planned building conflicts with the already-built furniture.

It may be possible to automate removal of the existing furniture, designation of the floor, and replacement of the furniture afterwards, but that would be a very advanced tool indeed (it's actually planned for `quickfort` to be able to do things like this so you can automatically apply the next blueprint when jobs associated with the previous one are completed, but again, big task, not yet done)

**Reply by adiamaq** [reply]
Awesome, I understand. Thank you again, and for all the work you guys are doing! <3

**Comment by jtuu** [comment]
Any update on this?
